### ISSUE TITLE
Address information in Location (Bugzilla Bug 348)

### ISSUE THREAD

**Issue by datexii** [DATEX-II-EU/DatexII#348]
![LocationExtension](https://github.com/DATEX-II-EU/DatexII/assets/75827629/a395b643-4edb-470d-8e0e-9d4a05d371d5)
This issue was created automatically with bugzilla2github.py
# Bugzilla Bug 348
Date: 2021-05-12T08:12:42+02:00
From: @JoergFst
To:   Jean-Philippe Mechine &lt;<Jean-Philippe.Mechin@cerema.fr>&gt;

Last updated: 2021-05-12T08:12:42+02:00

**Comment by LBlaive** [comment]
This issue needs to be related to issue no 448 below. Such proposal would comply with the rule stated for DATEX II to only use standardised location referencing systems. As EN ISO 19160-4 has been a published standard since 2023 it looks fine to integrate this proposal, especially in an urban context (standalone or in combination with linear system, i.e. a postal address used as referent) where the other location referencing systems are less convenient. 
The assessment of this proposal as regarding the published version of EN ISO 19160-4:2023 shows some differences:

- The proposal only deals with the delivery point and not with the addressee/mailee whereas the standard encompasses both specifications;
- Whereas EN ISO 19160-1:2015 is more global dealing with addresses in general with a temporal dimension, EN ISO 19160-4 is focused on postal addresses (based on UPU structures). This proposal seems in-between when detailin "AddressLines" but actually element in the corresponding enumeration;
- The "timeZone" attribute is foreign to a location reference definition and should be moved outside;
- It is not clear if the managed components of an address are lines or elements;
- There are some differences in terminology like e.g. "street' in the proposal instead of "thoroughfare" in the standard, "PO box" instead "delivery service identifier", "houseNumber" instead of "premises identifier" or "apartment" instead of "door";
- Some values are also missing ("wing", "stairwell, "delivery service qualifier")
- It seems usable for point locations directly. For linear locations the best use is consider such addresses as referent for a "LinearWithinLinearElement" instance.

The proposal is the following:

- Creation of a new package "PostalAddress" in LocationReference"
- Renaming of the "Address", "AddressLine" and "AddressLineTypeEnum" classes as "PostalAddress", "AddressComponent" and "AddressComponentTypeEnum"
- Addition of the missing attributes into these classes;
- Addition and renaming of the literals in the "AddressComponentTypeEnum" to better abide by EN ISO 19160-4;
- The "PostalAddress" is associated to the "PointLocation" class (multiplicity 0..1) and the corresponding is alos linked to this new association;
- As "PostalAddress" is a class, it is proposed to reuse it for creating a new type of referent.

![PostalAddress_as_LocationPoint](https://github.com/user-attachments/assets/4bd5800d-e327-4905-886e-12d1e36a2f62)

**Comment by werkenr** [comment]
Thoroughfare is not part of the AddressComponentTypeEnum, how to describe my streetname?
Is thoroughfare the correct term, since a dead-end street is not a thoroughfare.

**Comment by LBlaive** [comment]
That is true "thoroughfare" was missing in the first version of the proposed package. It is now corrected (see below).
![PostalAddress_as_LocationPoint_v4](https://github.com/user-attachments/assets/5bc83f69-c7ef-4ed4-ac95-22b100a0950e)

This new proposal tries to be closer of EN ISO 19160-4:2023 than the initial proposal even it has introduced some simplification in the instantiation of each address components (here a thoroughfare instance is e.g. "Rue d'Alésia" and not composed of the three elements "Rue", "d'Alésia" and "prefix" (to express "Rue" is before "d'Alésia")

Note: "Thoroughfare" is the term used in EN ISO 19160-4:2023 (which stems form a UPU standard) to designate any road element including cul-de-sac (or dead ends).
The corresponding definition in EN ISO 19160-4:2023 (p. 18) is: "Element which identifies the road or part of a road or other access route along which a delivery point can be accessed, either directly or via a secondary or tertiary road or access route."

Question: EN ISO 19148-4 allows considering up to three thoroughfares in a hierarchy as expressed in the definition above. The optional "Level" attribute is used for this specific case.

**Comment by LBlaive** [comment]
That is true "thoroughfare" was missing in the first version of the proposed package. It is now corrected (see below).
![PostalAddress_as_LocationPoint_v4](https://github.com/user-attachments/assets/5bc83f69-c7ef-4ed4-ac95-22b100a0950e)

This new proposal tries to be closer of EN ISO 19160-4:2023 than the initial proposal even it has introduced some simplification in the instantiation of each address components (here a thoroughfare instance is e.g. "Rue d'Alésia" and not composed of the three elements "Rue", "d'Alésia" and "prefix" (to express "Rue" is before "d'Alésia")

Note: "Thoroughfare" is the term used in EN ISO 19160-4:2023 (which stems form a UPU standard) to designate any road element including cul-de-sac (or dead ends).
The corresponding definition in EN ISO 19160-4:2023 (p. 18) is: "Element which identifies the road or part of a road or other access route along which a delivery point can be accessed, either directly or via a secondary or tertiary road or access route."

Question: EN ISO 19148-4 allows considering up to three thoroughfares in a hierarchy as expressed in the definition above. The optional "Level" attribute is used for this specific case (the accepted values are 1 to 3).

**Comment by iancornwell1** [comment]
I am concerned that Level might be might be misinterpreted in one or two different ways. I respect the diligent representation of the 19160 standard, but do we need that extra functionality for the secondary or tertiary access roads? If we do need that functionality, then I think I would prefer one of the two following solutions without the  level attribute: (i) only allow these extra access roads in a specialised subclass of PostalAddressComponent that is constrained to be of type thoroughfare and has optional secondary and tertiary access roads (ii) keep a single class, without level, and define in a note that when there are two or three thoroughfares with consecutive "order", they should be understood to represent secondary and tertiary access roads. Normally I prefer using design features to using constraints in notes, but since this feature may be rare, I think I prefer (ii).

**Comment by LBlaive** [comment]
I also wondered if we needed this possibility in a pure European context. EN ISO 19160-4 is first an ISO standard that also takes into consideration non-European needs like the interweaving of several region levels for Korean or Japanese addresses.
As I have not got any answer I prefer to keep the possibility to define different levels of thoroughfare. I agree that the misuse of the "Level" attribute is to avoid and I am content to remove the attribute from the model and choose the solution (ii).
Any idea about the constraint (?) note? Proposal: "In case it is necessary to define secondary and possibly tertiary access roads as part of a thoroughfare, they shall be defined as separate thoroughfares in an order representing respectively the primary, the secondary and the tertiary access roads when they exist."

**Comment by iancornwell1** [comment]
The call 21/11 discussed whether we should use the term "thoroughfare" and ran out of time to conclude. I confirmed that in normal English usage it does imply that there should an open path and not a dead end, and I have seen signs in UK that warn "no thoroughfare". So it is not ideal, but we were shown that is the term used in that EN with a defined meaning that would even include dead ends. The EN seems flawed, but it is the EN. We discussed using another term with clarification that it corresponds to the 19160 "thoroughfare", or the other way round - keep "thoroughfare" and clarify that it in this context it is from the EN and not restricted to be a through-road. If we used another term, what it would be? The candidates I can think of are "road name" and "street", each of which has its drawbacks. I say "road name" because "road" may suggest number, which would not normally be used in a postal address. Probably the most natural in the context of postal address is "street", but ultimately I could accept any of "thoroughfare", "street" or "road name", as long as the definition clarifies.
